beard/clustering/blocking_funcs.py

```diff
@@ -0,0 +1,487 @@
+        is counted FULL_NAME_MATCH_BONUS times to the result, as it indicates
+        the same author with bigger probability.
+        + Initials match if none of compared elements is a full name.
+        + Initials match if one od the compared elements is a full name.
+
+        Two characters in initials - ``A`` and ``H`` are handled in special
+        way in initial comparison, as the double metaphone can output an empty
+        string for them.
+
+        Parameters
+        ----------
+        :param new_first_names: tuple
+            Strings representing given names of the new author
+        :param last_name: tuple
+            Tokens, usually one, representing last surname(s) of the author.
+
+        Raises
+        ------
+        :raises: KeyError
+            When the last name is not included in the cluster
+
+        Returns
+        -------
+        :returns: integer
+            Score for matching first names in the cluster.
+        """
+        result = 0
+        if last_name in self._content:
+            for first_names, occurences in \
+                    six.iteritems(self._content[last_name]):
+                first_names_length = len(first_names)
+                old_names_index = 0
+                names_match = self.FULL_NAME_MATCH_BONUS
+                for new_name in new_first_names:
+                    while old_names_index < first_names_length:
+                        full_names_match = \
+                            self._names_match(first_names[old_names_index],
+                                              new_name)
+                        if not full_names_match:
+                            # There was no match, check next name from the
+                            # old signature
+                            old_names_index += 1
+                        else:
+                            if names_match:
+                                names_match = full_names_match
+                            # Go to the next name from the new ones.
+                            break
+                    if old_names_index == first_names_length:
+                        # No match.
+                        names_match = 0
+                        break
+                if names_match:
+                    result += occurences * names_match
+            return result
+        self._raise_keyerror(last_name)
+
+    def single_surname_signatures(self):
+        """Return the number of signatures with only one surname.
+
+        The only exception is that when the block was created by a
+        signature which consisted of more than one surname, the result will
+        be increased by one, so that the result can be used as a denominator.
+
+        Returns
+        -------
+        :returns: integer
+            Number of signatures with only one surname or 1 if there are no
+            such.
+        """
+        return self._single_surname_signatures
+
+    def _names_match(self, name1, name2):
```

Comment:
Assuming `FULL_NAME_MATCH_BONUS` is factored out, this can be a standalone function. No need to make it a method